Objects/longobject.c

```diff
@@ -5260,6 +5260,40 @@ long_is_finite(PyObject *v)
 }
 #endif
 
+/*[clinic input]
+int.as_integer_ratio
+
+Return integer ratio.
+
+Return a pair of integers, whose ratio is exactly equal to the original int
+and with a positive denominator.
+
+Raise OverflowError on infinities and a ValueError on NaNs.
+
+>>> (10).as_integer_ratio()
+(10, 1)
+>>> (0).as_integer_ratio()
+(0, 1)
+>>> (11).as_integer_ratio()
+(11, 1)
+>>> (-10).as_integer_ratio()
+(-10, 1)
+[clinic start generated code]*/
+
+static PyObject *
+int_as_integer_ratio_impl(PyObject *self)
+/*[clinic end generated code: output=e60803ae1cc8621a input=ce9c7768a1287fb9]*/
+{
+    PyObject *denominator = NULL;
+    PyObject *result_pair = NULL;
+
+    denominator = PyLong_FromLong(1);
+    result_pair = PyTuple_Pack(2, self, denominator);
```

Comment:
If `PyLong_FromLong` returns `NULL` then `PyTuple_Pack` and `Py_DECREF(denominator)` will fail.

Reply:
I think the whole function can be replaced with `return PyTuple_Pack(2, self, _PyLong_One)`.

Reply:
@pablogsal But PyLong_FromLong always receive the 1. How could it fail?

Reply:
The contract of `PyLong_FromLong` says that it can return `NULL` on failure. Even if the current implementation (that uses the array of small ints) makes it improbable/impossible to fail, this can change in the future and the function will fail.